packages/@aws-cdk/aws-cloudwatch/lib/dashboard.ts

```diff
@@ -33,7 +33,7 @@ export class Dashboard extends Construct {
             dashboardBody: new Token(() => {
                 const column = new Column(...this.rows);
                 column.position(0, 0);
-                return tokenAwareJsonify({ widgets: column.toJson() });
+                return JSON.stringify({ widgets: column.toJson() });
```

Comment:
It would be ideal if we can get rid of `tokenAwareJsonify`, but I am wondering if the new approach takes care of tokens that resolve to strings and may contain characters that require JSON escaping performed by [`stringifyStrings`](https://github.com/awslabs/aws-cdk/pull/518/files#diff-1b122baf1dfff6e35b392172e7cbf6b3L40)

Reply:
Okay, it depends a little on what we want here. For example, do we want to strip `undefined`s from stringified objects if it turns out that the `Token` in there evaluates to undefined? For example, would you expect this:

```ts
const obj1 = {
    literal: "okay",
    gone: new Token(() => undefined),
}


const obj2 = {
    literal: "okay",
    gone: undefined,
}

// true or no?
resolve(JSON.stringify(obj1)) == resolve(JSON.stringify(obj2))
```

If so, then we're going to need to delay the `JSON.stringify()` operation until `resolve()`, which means that we need `tokenAwareJsonify()` after all, to wrap the stringification operation in a Token itself.

Reply:
For my part, I think we WOULD want the equivalence that I posited. Otherwise, users have to be aware where in a CloudFormation template their property values are used which is exactly the kind of the thing we're trying to get rid of.

Reply:
Here is the test that covers the case I was talking about: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/cdk/test/test.token-aware-jsonify.ts#L60

Reply:
Oh sorry, I know, I got it. While working on this I just thought of some more edge cases we might want to cover, though. Sorry for not being more clear.